File: #0268.missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {

        int size = nums.size();
        int result = 0;
        for (int i = 0; i < size; i++) {
            result ^= nums[i];
            result ^= i;
        }
        return result ^= size;
    }
};

File: #0268.missing-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    const results = new Array(nums.length+1).fill(-1);
    for (const num of nums) {
        results[num] = num;
    }
    return results.indexOf(-1);
};
